command/rootchain/deploy/deploy.go

```diff
@@ -478,6 +485,57 @@ func populateExistingTokenAddr(eth *jsonrpc.Eth, tokenAddr, tokenName string,
 	return nil
 }
 
+// registerChainOnStakeManager registers child chain and its supernet manager on rootchain
+func registerChainOnStakeManager(txRelayer txrelayer.TxRelayer,
+	rootchainCfg *polybft.RootchainConfig, deployerKey ethgo.Key) (int64, error) {
+	registerChainFn := &contractsapi.RegisterChildChainStakeManagerFn{
+		Manager: rootchainCfg.CustomSupernetManagerAddress,
+	}
+
+	encoded, err := registerChainFn.EncodeAbi()
+	if err != nil {
+		return 0, fmt.Errorf("failed to encode parameters for registering child chain on supernets. error: %w", err)
+	}
+
+	txn := &ethgo.Transaction{
+		To:    (*ethgo.Address)(&rootchainCfg.StakeManagerAddress),
+		Input: encoded,
+	}
+
+	receipt, err := sendTransaction(txRelayer, txn, checkpointManagerName, deployerKey)
+	if err != nil {
+		return 0, err
+	}
+
+	var (
+		childChainRegisteredEvent contractsapi.ChildManagerRegisteredEvent
+		found                     bool
+		chainID                   int64
+	)
+
+	for _, log := range receipt.Logs {
+		doesMatch, err := childChainRegisteredEvent.ParseLog(log)
+		if !doesMatch {
+			continue
+		}
+
+		if err != nil {
+			return 0, err
+		}
```

Comment:
Nitpick. I'm aware that `ParseLog` first returns false if signature is not matched, but anyways it is better to first check if there are errors, just in case.
```suggestion
		if err != nil {
			return 0, err
		}

        if !doesMatch {
			continue
		}
```

Reply:
Done: https://github.com/0xPolygon/polygon-edge/pull/1406/commits/e747436ded2572c7cb896cb1deb8daaccc4b451a